alma/lib/Helpers/OrderStateHelper.php

```diff
@@ -36,42 +38,49 @@
 class OrderStateHelper
 {
     /**
-     * @var \Context|null
+     * @var int
      */
-    protected $context;
+    protected $contextLanguageId;
+
+    /**
+     * @var
+     */
+    protected $orderStateFactory;
 
     /**
      * @param ContextFactory $contextFactory
-     * @codeCoverageIgnore
+     * @param OrderStateFactory $orderStateFactory
      */
-    public function __construct($contextFactory)
+    public function __construct($contextFactory, $orderStateFactory)
     {
-        $this->context = $contextFactory->getContext();
+        try {
+            $this->contextLanguageId = $contextFactory->getContextLanguageId();
+        } catch (AlmaException $e) {
+            $this->contextLanguageId = null;
```

Comment:
Language id can't be null, Maybe we can return the default id language ?
With something like this ? `Configuration::get('PS_LANG_DEFAULT', null, $shop->id_shop_group, $shop->id);`

Reply:
This is unit tests, if we follow, the documentation for Prestatshop, it can be null (of course it should not be because it will means that the merchant website has an issue), and if it is it's an error in our case we need to stop the executtion, we do not want a fallback

Reply:
My suggestion was to avoid any exceptions on our side. Because this shouldn't happen in a case where there is no language id and in this case it's because the merchant doesn't have a language which would be blocking him before our module.
But it shouldn't happen if the merchant's environment is stable :)

Reply:
I can explain it you want a meet :)

Reply:
Ok fine see in call